lms/services/canvas_api/client.py

```diff
@@ -299,13 +299,31 @@ class _ListGroupCategories(RequestsResponseSchema):
         id = fields.Integer(required=True)
         name = fields.Str(required=True)
 
-    def group_category_groups(self, group_category_id):
-        """List groups that belong to the group category/group set `group_category_id`."""
-        return self._client.send(
+    def group_category_groups(self, course_id, group_category_id):
+        """
+        List groups that belong to the group category/group set `group_category_id`.
+
+        :param course_id: Course canvas ID.
+            Canvas group_categories are in an unique namespace for all courses
+            so the API call doesn't take the `course_id` but the response does
+            include to which course they belong. We wan't to make sure they belong the right course.
+        :param group_category_id: Group category ID to list the groups from.
+
+        :raise CanvasAPIError: When group_category_id doesn't belong to course_id.
+        """
+        groups = self._client.send(
             "GET",
             f"group_categories/{group_category_id}/groups",
             schema=self._ListGroups,
         )
+        # The response might include groups from a course other than the current one.
+        # This is most likely due to course copy, the groups are from the original course.
+        if any((group["course_id"]) != int(course_id) for group in groups):
```

Comment:
>  or it could be because the group set has been deleted.

Not exactly, will get an error before in that case, getting a group back from the API and that group being from another course will only happen in the context of course copy IMO.

Reply:
Tweaked your suggestion to accommodate that.